Realm/Realm/Linq/WhereClauseVisitor.cs

```diff
@@ -0,0 +1,358 @@
+                        throw new NotSupportedException(binaryLeftMemberExpression + " is null or not a supported node type.");
+                    }
+                }
+
+                if (binaryExpression.Left is ConstantExpression binaryLeftConstantExpression)
+                {
+                    if (binaryLeftConstantExpression.Value == null)
+                    {
+                        comparisonNode.Left = new ConstantNode()
+                        {
+                            Value = binaryLeftConstantExpression.Value,
+                            Type = "string"
+                        };
+                    }
+                    else
+                    {
+                        comparisonNode.Left = new ConstantNode()
+                        {
+                            Value = binaryLeftConstantExpression.Value,
+                            Type = GetKind(binaryLeftConstantExpression.Value.GetType())
+                        };
+                    }
+                }
+
+                if (binaryExpression.Right is MemberExpression binaryRightMemberExpression)
+                {
+                    if (binaryRightMemberExpression.Expression != null && binaryRightMemberExpression.Expression.NodeType == ExpressionType.Parameter)
+                    {
+                        comparisonNode.Right = new PropertyNode()
+                        {
+                            Name = GetColumnName(binaryRightMemberExpression, binaryRightMemberExpression.NodeType),
+                            Type = GetKind(binaryRightMemberExpression.Type)
+                        };
+                    }
+                    else
+                    {
+                        throw new NotSupportedException(binaryRightMemberExpression + " is null or not a supported node type.");
+                    }
+                }
+
+                if (binaryExpression.Right is ConstantExpression binaryRightConstantExpression)
+                {
+                    if (binaryRightConstantExpression.Value == null)
+                    {
+                        comparisonNode.Right = new ConstantNode()
+                        {
+                            Value = binaryRightConstantExpression.Value,
+                            Type = "string"
+                        };
+                    }
+                    else
+                    {
+                        comparisonNode.Right = new ConstantNode()
+                        {
+                            Value = binaryRightConstantExpression.Value,
+                            Type = GetKind(binaryRightConstantExpression.Value.GetType())
+                        };
+                    }
+                }
+
+                returnNode = comparisonNode;
+            }
+
+            return RealmLinqExpression.Create(returnNode);
+        }
+
+        protected override Expression VisitUnary(UnaryExpression node)
+        {
+            NegationNode returnNode = new NegationNode();
```

Comment:
Here you are creating already a node that is very specific. We should keep it as general until we can get the info on the specific type. I know we have only one at the moment, but this could grow